src/components/organisms/Settings/Project/Dataset/hooks.ts

```diff
@@ -38,9 +34,11 @@ export default (projectId: string) => {
     notifyOnNetworkStatusChange: true,
   });
 
-  const nodes = data?.datasetSchemas.edges.map(e => e.node) ?? [];
-
-  const datasetSchemas = nodes.filter(Boolean) as DatasetSchemas;
+  const datasetSchemas = data?.datasetSchemas.edges
+    .map<Item | undefined>(item =>
+      item.node ? { id: item.node.id, name: item.node.name } : undefined,
+    )
+    .filter((item): item is Item => !!item);
```

Comment:
Same as above: destructure the node and then you don't need to assign the values to keys. And again, not sure if the filter is necessary? Let me know